src/EntityGraphQL/Compiler/GqlNodes/GraphQLInlineFragmentField.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using EntityGraphQL.Compiler.Util;
+using EntityGraphQL.Schema;
+
+namespace EntityGraphQL.Compiler
+{
+    public class GraphQLInlineFragmentField : BaseGraphQLField
+    {
+        public GraphQLInlineFragmentField(ISchemaProvider schema, string name, Expression? nodeExpression, ParameterExpression rootParameter, IGraphQLNode parentNode)
+            : base(schema, null, name, nodeExpression, rootParameter, parentNode, null)
+        {
+        }
+
+        public override void AddField(BaseGraphQLField field)
+        {
+            field.Name = $"{Name}.{field.Name}";
```

Comment:
Why is this needed?

Reply:
This is because multiple 'subtypes' might have a field of the same name but different expressions, so this prevents them overwriting each other (it is removed later during actual selection).  Could have potentially merged the expressions instead using a IS check, but was difficult to get it working at that spot in the code.

Reply:
Can you point me to where this is undone?

Reply:
added a comment to the line that undoes it